scripts/gulpfiles/test_tasks.js

```diff
@@ -371,6 +370,16 @@ function advancedCompileInBrowser() {
   return runTestTask('advanced_compile_in_browser', runCompileCheckInBrowser);
 }
 
+/**
+ * Verify the built Blockly type definitions compile with the supported
+ * TypeScript examples included in `./tests/typescript`.
+ * @returns {Promise} Asynchronous result.
+ */
+function typeDefinitions() {
+  return runTestCommand('type_definitions',
+    'tsc -p ./tests/typescript/tsconfig.json');
```

Comment:
Please write output to somewhere in `build/` rather than in `tests/`.

I think it would be preferable to do so using `-outDir` and `path.join(BUILD_DIR, 'tests' /* optionally more */)` as `buildJavaScript` (in `build_tasks.js`) does.